torch_geometric/datasets/web_qsp_dataset.py

```diff
@@ -0,0 +1,359 @@
+
+        # Placeholder for storing the embeddings
+        all_embeddings_list = []
+
+        # Iterate through batches
+        with torch.no_grad():
+
+            for batch in dataloader:
+                # Move batch to the appropriate device
+                batch = {key: value.to(device) for key, value in batch.items()}
+
+                # Forward pass
+                embeddings = model(input_ids=batch["input_ids"],
+                                   att_mask=batch["att_mask"])
+
+                # Append the embeddings to the list
+                all_embeddings_list.append(embeddings)
+
+        # Concatenate the embeddings from all batches
+        all_embeddings = torch.cat(all_embeddings_list, dim=0).cpu()
+    except:  # noqa
+        print(
+            "SBERT text embedding failed, returning torch.zeros((0, 1024))...")
+        return torch.zeros((0, 1024))
+
+    return all_embeddings
+
+
+class WebQSPDataset(InMemoryDataset):
+    r"""The WebQuestionsSP dataset was released as part of
+    “The Value of Semantic Parse Labeling for Knowledge
+    Base Question Answering”
+    [Yih, Richardson, Meek, Chang & Suh, 2016].
+    It contains semantic parses, vs. answers, for a set of questions
+    that originally comes from WebQuestions [Berant et al., 2013]."
+    Processing based on "G-Retriever: Retrieval-Augmented Generation
+    for Textual Graph Understanding and Question Answering".
+    Requires datasets and transformers from HuggingFace.
+
+    Args:
+        root (str): Root directory where the dataset should be saved.
+        force_reload (bool, optional): Whether to re-process the dataset.
+            (default: :obj:`False`)
+    """
+    def __init__(
+        self,
+        root: str = "",
+        force_reload: bool = False,
+    ) -> None:
+        missing_imports = False
+        missing_str_list = []
+        if not WITH_PCST:
+            missing_str_list.append('pcst_fast')
+            missing_imports = True
+        if not WITH_TRANSFORMERS:
+            missing_str_list.append('transformers')
+            missing_imports = True
+        if not WITH_DATASETS:
+            missing_str_list.append('datasets')
+            missing_imports = True
+        if not WITH_PANDAS:
+            missing_str_list.append('pandas')
+            missing_imports = True
+        if missing_imports:
+            missing_str = ' '.join(missing_str_list)
+            error_out = f"`pip install {missing_str}` to use this dataset."
+            raise ImportError(error_out)
+        self.prompt = "Please answer the given question."
+        self.graph = None
+        self.graph_type = "Knowledge Graph"
+        self.model_name = "sbert"
+        self.device = torch.device(
+            "cuda" if torch.cuda.is_available() else "cpu")
+        super().__init__(root, None, None, force_reload=force_reload)
+        self.load(self.processed_paths[0])
+
+    @property
+    def raw_file_names(self) -> List[str]:
+        return []
```

Comment:
Is there a good reason this is empty?

Reply:
i left it blank since the raw files are handled by HuggingFace datasets:
```
    @property
    def raw_file_names(self) -> List[str]:
        return []

    @property
    def processed_file_names(self) -> List[str]:
        return ["list_of_graphs.pt", "pre_filter.pt", "pre_transform.pt"]

    def download(self) -> None:
        dataset = datasets.load_dataset("rmanluo/RoG-webqsp")
        self.raw_dataset = datasets.concatenate_datasets(
            [dataset["train"], dataset["validation"], dataset["test"]])
        self.split_idxs = {
            "train":
            torch.arange(len(dataset["train"])),
            "val":
            torch.arange(len(dataset["validation"])) + len(dataset["train"]),
            "test":
            torch.arange(len(dataset["test"])) + len(dataset["train"]) +
            len(dataset["validation"])
        }
```

i am open to suggestions